File: client/src/types/dish.ts
/**
 * 菜品类型（2026-09-22 列表 / 详情出参拆分，见 docs/feature/client-首页菜品浏览.md D 项）
 *
 * **列表行 `DishListItem`**（`GET /dishes`，`DishListItemVO` **恰为 8 字段**）：
 *   id / name / coverImage / price / originalPrice / avgRating / canteenName / stallName
 * **详情 `DishDetail`**（`GET /dishes/{id}`，`DishDetailVO` 15 字段 + `ratingDistribution`）：
 *   id / name / price / originalPrice / description / images / stallName / canteenName /
 *   floor / avgRating / ratingCount / dietType / ingredients / flavorTags / serveTemp
 *
 * 端上定型说明：
 * - 金额一律「元」：API 层由分转元（`fenToYuan`），视图层直取展示；
 * - `canteenName` → 端上别名 `canteen`，`avgRating` → `rating`；
 * - 列表**不含**详情专属字段（`description` / `images` / `floor` / `ratingCount` / 四维）；
 *   列表图片只给 `coverImage`（后端首图绝对 URL；无图空串）——列表**不得回流** `images` 数组；
 * - 描述四维**下发机器值**（`dietType` / `serveTemp` 单值，`ingredients` / `flavorTags` 为数组）；
 *   中文展示值由**四维字典端点**提供（见 `api/dish-attribute.ts` + `stores/dish-attribute.ts`），
 *   端上 SHALL NOT 硬编码「机器值 → 中文」映射表（2026-09-23 §7.40 R4）。
 *
 * 已全链删除（SHALL NOT 回流）：promoPrice / status / createdAt / canteenId / stallId /
 * viewCount / tags / spiceLevel / region / windowNo / updatedAt / latitude / longitude /
 * distance（坐标与距离随「计算距离」概念整体下线）。
 */
export interface DishListItem {
  id: number
  name: string
  /** 现价（元；API 层已由分转元）。**价格展示唯一数据源** */
  price: number
  /** 原价（元，可空）；有值且大于 price 时端上在原价加删除线表示折扣 */
  originalPrice?: number
  /** 封面首图（后端 `coverImage`，列表唯一图片字段；无图空串） */
  coverImage: string
  /** 平均评分（后端 avgRating，口径 = 仅未隐藏评价） */
  rating: number
  /** 食堂名称（后端 canteenName） */
  canteen: string
  /** 档口名称 */
  stallName: string
}

interface RatingDistribution {
  star: number
  count: number
}

/** 详情（`GET /dishes/{id}`）：列表 8 字段之外，额外含详情专属字段与评分分布 */
export interface DishDetail {
  id: number
  name: string
  price: number
  originalPrice?: number
  description: string
  /** 多图 URL 数组（详情专属） */
  images: string[]
  /** 首图派生字段（= images[0]，详情图集展示便利；非后端出参） */
  image: string
  rating: number
  ratingCount: number
  canteen: string
  stallName: string
  /** 档口所属楼层（如 1F/2F；详情专属） */
  floor?: string
  /** 描述四维·荤素（**机器值**：meat / half / veg / halal；中文由字典提供） */
  dietType?: string
  /** 描述四维·主料（**机器值数组**：如 `['chicken','rice']`；中文由字典提供） */
  ingredients?: string[]
  /** 描述四维·口味（**机器值数组**：如 `['spicy','sour']`；吸收原「辣度」语义，中文由字典提供） */
  flavorTags?: string[]
  /** 描述四维·冷热（**机器值**：hot / room / ice；中文由字典提供） */
  serveTemp?: string
  ratingDistribution: RatingDistribution[]
}

/**
 * 列表查询参数（`GET /dishes`，**完整参数集恰为 4 项**：page / pageSize / keyword / mealType）。
 * <p>
 * 2026-09-22：`canteenId` / `minPrice` / `maxPrice` 随「食堂 / 价格筛选全量下线」删除
 * （SHALL NOT 回流）；`sortBy` / `sortOrder` 已于 2026-09-21 收敛（排序恒为服务端热度倒序）。
 */
export interface DishQuery {
  keyword?: string
  /** 菜品大类筛选（首页横向标签栏；值为大类枚举键；不传 = 全部） */
  mealType?: string
  page?: number
  pageSize?: number
}

/**
 * 猜你喜欢词（GET /dishes/for-you）。
 * 2026-09-22 change search-page-refresh：由原「热搜词」类型改名——语义由
 * 「热度派生热搜词」变为「**随机抽取在售菜品名**」（不看热度、不排序、不做个性化，故不缓存）；
 * **契约留扩展位**：将来升级为个性化 / 推荐算法时仍为 `keyword` 列表，端上无需改造。
 */
export interface GuessLike {
  keyword: string
}

/**
 * 菜品大类字典项（`GET /dishes/meal-types`）：
 * 文案 / 顺序 / 子集全由后端下发（空类自动隐藏），**端上不得维护任何中文映射**。
 */
export interface MealType {
  key: string
  label: string
  order: number
}

File: server/src/main/java/com/bjtufood/dish/entity/Dish.java
package com.bjtufood.dish.entity;

import com.baomidou.mybatisplus.annotation.*;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 菜品实体类
 * <p>
 * 对应数据库表：dish
 * 价格以"分"为单位存储（如 12.00 元 = 1200 分），避免浮点精度问题。
 * 价格口径（2026-09-20 拍板 §7.26）：唯一数据源为 {@code price}（现价，已含折扣），
 * {@code originalPrice} 为可空原价；原「促销价」promoPrice 与列已全链删除。
 * 描述维度（2026-09-20 拍板 §7.28）：四维 dietType/ingredients/flavorTags/serveTemp 替换原 spiceLevel/region。
 */
@Data
@TableName("dish")
@Schema(description = "菜品")
public class Dish {

    @TableId(type = IdType.AUTO)
    @Schema(description = "菜品ID")
    private Long id;

    /** 所属档口ID */
    @Schema(description = "所属档口ID")
    private Long stallId;

    /** 菜品名称 */
    @Schema(description = "菜品名称", example = "牛肉拉面")
    private String name;

    /** 搜索别名（逗号分隔，管理员配置，可空）；搜索关键词命中别名也能找到该菜品 */
    @Schema(description = "搜索别名（逗号分隔，管理员配置，可空）", example = "拉面,牛肉面")
    private String alias;

    /** 现价（单位：分，已含折扣） */
    @Schema(description = "现价（分，已含折扣）", example = "1200")
    private Integer price;

    /** 原价（可空，单位：分）；originalPrice > price 视为有折扣 */
    @Schema(description = "原价（分，可空）", example = "1500")
    private Integer originalPrice;

    /** 菜品描述 */
    @Schema(description = "菜品描述")
    private String description;

    /** 菜品多图，JSON 字符串 */
    @Schema(description = "菜品多图JSON")
    private String images;

    /** 荤素/饮食属性（§7.28）：meat=荤 / half=半荤 / veg=素 / halal=清真（单选，可空） */
    @Schema(description = "荤素/饮食属性：meat=荤 / half=半荤 / veg=素 / halal=清真", example = "half")
    private String dietType;

    /** 主料/食材：逗号分隔机器值（§7.28，可空） */
    @Schema(description = "主料/食材（逗号分隔）：pork/beef/lamb/chicken/duck/fish/egg/tofu/mushroom/veg/noodle/rice", example = "chicken,rice")
    private String ingredients;

    /** 口味：逗号分隔机器值（§7.28，吸收原辣度语义，可空） */
    @Schema(description = "口味（逗号分隔）：spicy/numbing/sour/sweet/salty/umami/light/heavy", example = "spicy,sour")
    private String flavorTags;

    /** 冷热（§7.28）：hot=热食 / room=常温 / ice=冰（单选，可空） */
    @Schema(description = "冷热：hot=热食 / room=常温 / ice=冰", example = "hot")
    private String serveTemp;

    /**
     * 菜品大类（2026-09-21 §7.34，单值枚举）：set_meal 套餐盖饭 / stir_fry 家常小炒 /
     * noodle 面食粉类 / dry_pot 香锅干锅 / snack 风味小吃 / soup_drink 汤饮甜品（可空）。
     * 口径：一个菜品恰属一个大类；与属性维度（dietType/ingredients/flavorTags/serveTemp）分开；
     * 判定按「菜名与做法形态」。取值白名单见 {@code MealTypeConst}；**不进公开 DishVO 出参**。
     */
    @Schema(description = "菜品大类（单值）：set_meal 套餐盖饭 / stir_fry 家常小炒 / noodle 面食粉类 / dry_pot 香锅干锅 / snack 风味小吃 / soup_drink 汤饮甜品", example = "noodle")
    private String mealType;

    /** 状态：on（上架）/ off（下架）（菜品审核语义已整体退役，见 schema.sql dish 表注释） */
    @Schema(description = "状态", example = "on")
    private String status;

    // dish.reject_reason / dish.created_by 已于 2026-09-16 用户拍板「零消费即删除」退役：
    // reject_reason 恒 NULL（审核语义退役后无写入入口）、created_by 只写不读（upsert 留痕撤销）；
    // CREATE TABLE 已移除，存量库由 schema.sql 末尾 drop_zero_consumer_columns 幂等段清理。

    /** 浏览量 */
    @Schema(description = "浏览量")
    private Integer viewCount;

    /** 平均评分 */
    @Schema(description = "平均评分", example = "4.5")
    private BigDecimal avgRating;

    /** 评价数 */
    @Schema(description = "评价数")
    private Integer ratingCount;

    @TableField(fill = FieldFill.INSERT)
    @Schema(description = "创建时间")
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    @Schema(description = "更新时间")
    private LocalDateTime updatedAt;
}
